### DISCUSSION TITLE
Trigger websocket by another websocket

### DISCUSSION THREAD

**Discussion by andrew-c0** [fastapi/fastapi#6402] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
manager = ConnectionManager()

@app.websocket("/ws/chat/{room_id}/messages/{token}")
async def websocket_endpoint(room_id: int, token: str, websocket: WebSocket, db: Session = Depends(get_db)):
    user = await get_current_user(token, db)
    if user is not None:
        await manager.connect(websocket)
        try:
            while True:
                data = await websocket.receive_text()
                item_json = json.loads(data)
                item = schemas.Chat_RecordCreate(
                    user = item_json['user'],
                    content = item_json['content'],
                    content_type = item_json['content_type']
                )
                crud.add_chat_message(db, user, room_id, item)
                await manager.broadcast('{"status": "success", "data": "updated"}')
                #await websocket_unread_messages(token, websocket, db)
        except WebSocketDisconnect:
            manager.disconnect(websocket)

unreadChatManager = ConnectionManager()

@app.websocket('/ws/chat/unread_messages/{token}')
async def websocket_unread_messages(token: str, websocket: WebSocket, db: Session = Depends(get_db)):
    user = await get_current_user(token, db)
    if user is not None:
        if not unreadChatManager.isActive(websocket):
            await unreadChatManager.connect(websocket)
        try:
            unread_messages = crud.fetch_unread_messages(db, user)
            await unreadChatManager.broadcast('{"unread_messages":' + str(unread_messages) + '}')
        except WebSocketDisconnect:
            unreadChatManager.disconnect(websocket)
```

### Description

Hello. The issue at hand I think is more of how should it work rather that it's not working.
The idea is that I have a simple app, for chat. One websocket is used for real-time chatting between users and another is used for getting the unread messages through the rest of the app. My question is, how can I trigger the second websocket so I can have the unread messages updated in real time but updated only from the backend, using the first websocket?

I totally understand if my way of doing this sounds a bit off, but I'm more than open for different solutions. I want tought o avoid the frontend to trigger both websockets separately, since they should be working together, but updating separate contents of frontend.

### Operating System

Linux

### FastAPI Version

0.77.1

### Python Version

3.10.4

**Comment by jgould22** [comment, marked as answer]
You are essentially trying to implement pub/sub here where one socket is a subscription and the other a publication.

I suggest doing a little research on how that design pattern can be implemented as there are many ways to go about it.

If you care less about the implementation details then I would lean on an existing system that provides this functionality. There are plenty of options. One simple one is to use a Redis instance and have your python connect to it to use the pubsub functionality. 

https://redis.io/docs/manual/pubsub/

**Comment by andrew-c0** [comment]
Thanks for the answer! It's actually a good idea! For now, I'll close this issue.
